crates/sui-core/src/safe_client.rs

```diff
@@ -490,17 +490,14 @@ where
             .handle_transaction_info_request(request.clone())
             .await?;
 
-        let transaction_info = Transaction::new(transaction_info.transaction)
-            .verify()
-            .and_then(|verified_tx| {
-                self.check_transaction_info(
-                    &request.transaction_digest,
-                    verified_tx,
-                    transaction_info.status,
-                )
-            }).tap_err(|err| {
-                error!(?err, authority=?self.address, "Client error in handle_transaction_info_request");
-            })?;
+        let transaction = Transaction::new(transaction_info.transaction);
+        let transaction_info = self.check_transaction_info(
```

Comment:
I'm a little uncertain about this. Now this API does not return the user signature meaning and I could smell the risk in some scenarios.

Reply:
This function is `read`, so it does not deal with the write component. Let's say me as a business have tx_digest that a customer claimed to pay me the coffee money. Then I use this query to ask validators what tx_digest is about (I know I can ask a trusted fullnode but I'm particularly interested in "edge" case). Then the validator return some stuff to me, which has the validator quorums, but I don't verify the **user signature** at all before I let the customer grab and coffee and go. Would there be any problems with this?

Reply:
i think for the retail case, you would at least want to have a cert for the customers txn, otherwise you can’t tell if they are giving you an equivocating tx. (ideally you would have an effects cert) - there is possibly an edge case where you only have a tx cert, which is that if you receive the cert but can’t execute it before the epoch change, you may not be able to retry in the new epoch if you have a bad user sig.

However, my overall response is that:
- this is pretty low risk
- if you are are really worried, just don’t consider the sale closed until you have an fx cert, which for payments can be sub-second latency anyway, and
- i think the reality is that if we are going to have stateful authenticators of any kind, we cannot continue verifying transactions on the client until we have light client infrastructure
- a client who verifies user sigs and believes they are safe has a false sense of security anyway - until a cert is formed the tx on its own is meaningless, e.g. it may be equivocating